Replace documentation website with Docusaurus

GabLeRoux opened the pull request:
While working on replacing on #218 I found that there is a [docsearch `Docusaurus v2 template` available](https://docsearch.algolia.com/docs/templates#docusaurus-v2-template) and felt it would be interesting to give this optimized site generator a try. 

#### Changes

- [x] Replace documentation engine with https://docusaurus.io/
- [x] Upgrade to Node 14
- [x] Fix vercel deployment
- [x] Fix discord redirect
- [x] Remove blog entries until we have real blog entries
- [x] Add back home page
- [x] Setup typescript support again by following https://docusaurus.io/docs/typescript-support
- [x] Move `introduction` sidebar item at top of the list and have it link to desired platform (github or gitlab)
- [x] Integrate TailwindCSS, we can follow https://dev.to/sajclarke_62/using-tailwindcss-v3-in-docusaurus-in-5-steps-5c26
      see https://github.com/game-ci/documentation/pull/259#issuecomment-1118099725
- [x] Add back docker versions page (with login, etc.) See https://game.ci/docs/docker/versions
- [x] Fix small layout glitches in docker versions page
- [x] Fix dark theme in docker versions page
- [x] Integrate proper docs versioning using [docusaurus.io/docs/versioning](https://docusaurus.io/docs/versioning)
- [x] Add linting back, see https://github.com/game-ci/documentation/pull/259#discussion_r857174441
- [x] <del>Fix `Duplicate routes found!` warning:</del> See #266, for now I suggest we keep the warning until we get rid of AntD and move to an actual `mdx` docs page with a component that fits docs styling 👍 

```
[WARNING] Duplicate routes found!
- Attempting to create page at /docs/docker/versions, but a page already exists at this route.
This could lead to non-deterministic routing behavior.
```

This is due to a workaround to get the `versions` page to show in the navigation, but the page itself is handled by a docusaurus page.

---

For this deployment to work, vercel settings needs to be updated to `docusaurus 2`:

<img width="773" alt="image" src="https://user-images.githubusercontent.com/1264761/164992537-685c957f-d60f-4472-8bd5-f1e231018473.png">

⚠️ caution, this has an impact for other deployments. When working on this branch, update the value. When you're done, switch back to Next.

Concerning NodeJS version, it doesn't need to be updated as it can be determined from `package.json`'s value 👍. Docusaurus is not compatible with Node 12, but it is compatible with Node 14 and 16. Vercel is not compatible with Node 16 so we're sticking with Node 14 for now.


#### Checklist

<!-- please check all items and add your own -->

- [x] Read the contribution [guide](../blob/main/CONTRIBUTING.md) and accept the [code](../blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

GabLeRoux commented:
Tried deployment using `docusaurus 2` template in Vercel and getting some small issues.

<img width="1027" alt="image" src="https://user-images.githubusercontent.com/1264761/164992583-db563e57-f584-4ed7-8609-e8f24b2b41c2.png">

webbertakken commented on README.md:
````diff
@@ -1,21 +1,47 @@
+
+Using SSH:
+
+```bash
+USE_SSH=true yarn deploy
 ```
 
+Not using SSH:
+
+```bash
+GIT_USER=<Your GitHub username> yarn deploy
+```
````
These are not cross-platform compatible and we wouldn't deploy from local anyway. If anything this should probably move to `DEVELOPMENT.md`, as the readme is more for the public to see.

GabLeRoux replied:
This is docusaurus' default readme content, agreed about deployment, we're using vercel anyway for deployment

GabLeRoux replied:
I removed the `Deployment` section from readme (since we didn't have it before anyway). Only kept what's relevant

webbertakken commented on blog/2021-08-26-welcome/index.md:
```diff
@@ -0,0 +1,25 @@
+
+A blog post folder can be convenient to co-locate blog post images:
+
+![Docusaurus Plushie](./docusaurus-plushie-banner.jpeg)
```
While I like being able to add images we should probably make sure that we commit them to LFS from the start (and also squash this PR), to not slow down the repo by unnecessary deltas.

Suggest fixing that as part of this PR

webbertakken commented on docusaurus.config.js:
```diff
@@ -0,0 +1,116 @@
+              },
+            ],
+          },
+          // <ul id="item_3$Menu" class="ant-menu ant-menu-sub ant-menu-vertical" role="menu"><li class=" ant-menu-item-group"><div class="ant-menu-item-group-title" title="Projects">Projects</div><ul class="ant-menu-item-group-list"><li class="ant-menu-item" role="menuitem"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" class="anticon ant-menu-item-icon"  ...
```
Is this line intended? autogenerated somehow?

webbertakken commented on package.json:
```diff
@@ -1,107 +1,40 @@
-    "update-menu": "cross-env NODE_OPTIONS=\"-r esm -r dotenv/config\" ts-node -r tsconfig-paths/register tools/menu/cli-generate.ts",
-    "update-search": "cross-env NODE_OPTIONS=\"-r esm -r dotenv/config\" ts-node -r tsconfig-paths/register tools/search/push-search-payload.ts",
-    "analyze": "cross-env-shell NODE_OPTIONS=\"-r esm\" ANALYZE=true next build",
-    "lint": "prettier --write . --loglevel=warn && eslint . --quiet --fix",
-    "eslint": "eslint '*/**/*.{js,ts,tsx}' --quiet --fix",
```
We should keep the linting setup, unless there is a new setup that covers everything already.

webbertakken commented on package.json:
```diff
@@ -1,107 +1,40 @@
-    "fs": false,
-    "path": false
+  "engines": {
+    "node": "14.x"
```
lets go >=16.x which is latest LTS

webbertakken replied:
Any reason not to move the 16.x?

GabLeRoux replied:
Oh I thought I replied to that comment last time. Anyway, Vercel currently doesn't support Node 16:
https://github.com/vercel/community/discussions/37

It will be supported in the next few weeks according to last comment so we can update when it's supported :) I first started with node 16 but deployment failed.

GabLeRoux replied:
Node 16 support added 5 days ago, we can probably give `"node": "16.x"` a try 🚀

webbertakken commented on src/css/custom.css:
```diff
@@ -0,0 +1,45 @@
+/**
+ * Any CSS included here will be global. The classic template
+ * bundles Infima by default. Infima is a CSS framework designed to
+ * work well for content-centric websites.
+ */
```
Personally I would prefer to use `tailwind` over this if we have to redo styles either way, as it supports JIT compiling and it's one of the most popular and well maintained ones (and I'm using it on all my latest projects so I could make changes much faster in this repo then too)

webbertakken commented on tsconfig.json:
```diff
@@ -1,26 +0,0 @@
-{
-  "compilerOptions": {
-    "baseUrl": ".",
-    "target": "es5",
-    "lib": ["dom", "dom.iterable", "esnext"],
-    "allowJs": true,
-    "skipLibCheck": true,
-    "strict": false,
-    "forceConsistentCasingInFileNames": true,
-    "noEmit": true,
-    "esModuleInterop": true,
-    "module": "esnext",
-    "moduleResolution": "node",
-    "resolveJsonModule": true,
-    "isolatedModules": true,
-    "jsx": "preserve",
-    "paths": {
-      "@/components/*": ["components/*"],
-      "@/core/*": ["core/*"],
-      "@/logic/*": ["logic/*"],
-      "@/tools/*": ["tools/*"]
-    }
-  },
-  "exclude": ["node_modules"],
-  "include": ["next-env.d.ts", "@types/**/*", "**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
-}
```
Is there a replacement configuration or are we moving back to vanilla js?

GabLeRoux replied:
Very good question, I think we need to follow these docs:  
https://docusaurus.io/docs/typescript-support

webbertakken replied:
Yup sounds good. That should probably be in scope, as some things won't work without it I think.

webbertakken reviewed (approved):
This is a great idea!

webbertakken commented on vercel.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "redirects": [
+    {
+      "source": "/discord",
+      "destination": "https://discord.com/invite/WyPN5r9",
+      "permanent": false
+    }
+  ]
+}
```
Nice!

Just doublechecking: Is this automatically deployed every time?

GabLeRoux replied:
I think so, I tried on https://website-6tj5dd38a-game-ci.vercel.app/discord and it does the redirect properly 👍

webbertakken replied:
I checked it too, so you didn't use the vercel CLI manually at all?

GabLeRoux commented on .gitignore:
```diff
@@ -1,36 +1,22 @@
-.pnp.js
-
-# testing
-/coverage
```
I used the `.gitignore` generated by `docusaurus`, that's what happened. Here's a diff against the one from docusaurus and ours now:

```diff
0a1,2
> # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
> 
2a5,9
> /.pnp
> .pnp.js
> 
> # Testing
> /coverage
5a13,14
> /.search/*
> !/.search/.gitkeep
20a30,34
> 
> .DS_Store
> .idea
> .vscode
> .vercel
```

webbertakken replied:
Yes but the generated one doesn't include all the references that we need, for tests, the new yarn, specific editors etc.

The 5 lines in the first comment are what I think should be added into the generated one.

GabLeRoux replied:
@webbertakken Have a look at the latest changes to `.gitignore`, let me know if you think there's something missing, but it looks good to me now. 👍

webbertakken replied:
Looks mostly good now indeed.

See comment https://github.com/game-ci/documentation/pull/259/files#r857180824

GabLeRoux commented on .idea/inspectionProfiles/Project_Default.xml:
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="Eslint" enabled="true" level="WARNING" enabled_by_default="true" />
+  </profile>
+</component>
```
Nice catch thanks, I'll ignore .idea as well

webbertakken replied:
Yea, see previous comment for the complete list of things we do need and are not in the generated file ^^

webbertakken commented:
Fixes https://github.com/game-ci/documentation/issues/223

GabLeRoux commented on .nvmrc:
```diff
@@ -1 +0,0 @@
-v12.16.2
```
vercel doesn't support NodeJS 16 yet. So we must use NodeJS 14 for now. See https://github.com/vercel/community/discussions/37

I'm creating the `.nvmrc` file again with node 14 👍

webbertakken commented:
Closes https://github.com/game-ci/documentation/issues/130

webbertakken commented:
Fixes https://github.com/game-ci/documentation/issues/18

webbertakken commented on .gitignore:
```diff
@@ -1,36 +1,36 @@
+# Production
 /build
 /.search/*
 !/.search/.gitkeep
```
These two lines can be removed and this section can be merged with `Generated files`, rest looks good.

And I think we'll need LFS (`.gitattributes`) for the images. This one might be quite complete and suitable: https://gist.github.com/webbertakken/ff250a0d5e59a8aae961c2e509c07fbc

GabLeRoux commented:
I played a bit with tailwindcss and docusaurus, but now that I managed to do quite a few nice things, I realized that https://tailwindcss.com/docs/preflight removes headers and pages classic styling, which breaks the docs pages a little.

Before tailwindcss:
<img width="653" alt="CleanShot 2022-05-04 at 22 07 52@2x" src="https://user-images.githubusercontent.com/1264761/166854364-ca38c116-292c-4763-88a0-50a98d1854d8.png">

After tailwindcss:
<img width="653" alt="CleanShot 2022-05-04 at 22 08 04@2x" src="https://user-images.githubusercontent.com/1264761/166854374-22894c9e-ad0d-4ecf-992a-d7c6c4302a69.png">

I also confirmed that on [a blank docusaurus + tailwindcss project](https://docusaurus-tailwindcss-example-seven.vercel.app/) here.

Now tailwindcss has a [typography plugin](https://tailwindcss.com/docs/typography-plugin) used with the `prose` class which allows us to bring back good styling by default for titles, etc. but I can't figure out how to tell docusaurus to wrap docs page with a div.prose.

Then, reading the docs a bit more, I realized docusaurus uses its own [theme (infima)](https://infima.dev/). It's well documented and has examples we could easily use for home page:
https://infima.dev/docs/components/hero

So I'm thinking of removing tailwindcss again unless we find a way to bring back styling in docs pages. I really like to use tailwindcss, but I feel it'd be easier to just use [infima](https://infima.dev/).

🤔

GabLeRoux commented:
Ah, this might be useful: https://tailwindcss.com/docs/typography-plugin#changing-the-default-class-name

webbertakken commented on tailwind.config.js:
```diff
@@ -3,7 +3,5 @@ module.exports = {
   },
-  plugins: [
-    require('@tailwindcss/typography'),
-  ],
```
Typography might still be useful for other pages right? Or do all pages use the same theme as Docusaurus (is that a tailwind theme?).

GabLeRoux replied:
It's quite easy to add it back whenever we need it. And if we create pages using docusaurus, default's docusaurus theme will apply to these pages as well (ex; the blog that used to be there before I comment it for future use).

It's documented here:  
https://tailwindcss.com/docs/typography-plugin

The instructions are quite simple.

GabLeRoux replied:
I didn't need typography so not adding it if unused. We can consider it again if we needed later. Resolving conversation.

webbertakken commented:
I also found a request for official tailwind support (or supporting a tailwind based theme rather).

Seems like abstracting out the UI from the components is [a work in progress](https://github.com/facebook/docusaurus/issues/2961#issuecomment-1070793226), but it looks promising for the future.

Perhaps we can do without including the `base` from tailwind like you suggested and later upgrade to this tailwind-based theme from Docusaurus.

webbertakken commented on src/pages/docs/docker/versions.tsx:
```diff
@@ -0,0 +1,60 @@
+export default function VersionsPage(): JSX.Element {
+  const {siteConfig} = useDocusaurusContext();
+
+  const addCssToHead = () => {
+    const css = document.createElement('link');
+    css.rel = 'stylesheet';
+    css.href = 'https://unpkg.com/antd/dist/antd.css';
+    document.head.appendChild(css);
+  };
+
+  const removeCssFromHead = () => {
+    const css = document.querySelector('link[href="https://unpkg.com/antd/dist/antd.css"]');
+    if (css) {
+      css.remove();
+    }
+  };
+
+  React.useEffect(() => {
+    addCssToHead();
+    return () => {
+      removeCssFromHead();
+    };
+  }, []);
```
This can be safely removed when moving away from AntD

webbertakken reviewed (approved):
Looking great. Almost done it seems!

GabLeRoux commented on docs/04-github/01-getting-started.md:
```diff
@@ -46,12 +46,14 @@ Any subsequent steps assume you have read the above.
 ## Video Tutorial
 
-https://www.youtube-nocookie.com/embed/M2BZr02uai0
+<iframe src="//www.youtube-nocookie.com/embed/M2BZr02uai0" frameborder="0" allowfullscreen width="100%" height="600"></iframe>
```
I tried it in a different pr here:  
https://github.com/game-ci/documentation/pull/264

I'm able to get things working, but tried two methods so far and I can't get it to be correctly responsive so right now, the iframe is a better option until we figure out how to solve the styling issue.

webbertakken commented on versioned_docs/version-1/05-gitlab/01-getting-started.md:
```diff
@@ -0,0 +1,53 @@
+# Getting started
+
+This guide will demonstrate how to setup build and test automation for your Unity project hosted on gitlab using gitlab-ci.
```
Actually losing all version information from the docs is not desirable.

Could you make sure git recognises that it's the same file as before, by making changes in one commit and doing the rename in another?

GabLeRoux replied:
oh, hmm, not quite sure how to achieve this.  I used `yarn run docusaurus docs:version 1` command and replaced the copy of generated docs with the content of v1.

GabLeRoux replied:
From what I see, history looks fine. What you are looking at is an actual copy of the gitlab docs. V1 was only for github.

```diff
diff --git a/versioned_docs/version-1/03-docker/versions.md b/versioned_docs/version-1/03-docker/versions.md
new file mode 100644
index 0000000..6af101f
--- /dev/null
+++ b/versioned_docs/version-1/03-docker/versions.md
@@ -0,0 +1 @@
+[//]: # ( See src/pages/docs/docker/versions.tsx )
diff --git a/docs/04-github/v1/01-getting-started.md b/versioned_docs/version-1/04-github/01-getting-started.md
similarity index 100%
rename from docs/04-github/v1/01-getting-started.md
rename to versioned_docs/version-1/04-github/01-getting-started.md
diff --git a/docs/04-github/v1/02-activation.md b/versioned_docs/version-1/04-github/02-activation.md
similarity index 100%
rename from docs/04-github/v1/02-activation.md
rename to versioned_docs/version-1/04-github/02-activation.md
diff --git a/docs/04-github/v1/03-test-runner.md b/versioned_docs/version-1/04-github/03-test-runner.md
similarity index 100%
rename from docs/04-github/v1/03-test-runner.md
rename to versioned_docs/version-1/04-github/03-test-runner.md
diff --git a/docs/04-github/v1/04-builder.md b/versioned_docs/version-1/04-github/04-builder.md
similarity index 100%
rename from docs/04-github/v1/04-builder.md
rename to versioned_docs/version-1/04-github/04-builder.md
diff --git a/docs/04-github/v1/05-returning-a-license.md b/versioned_docs/version-1/04-github/05-returning-a-license.md
similarity index 100%
rename from docs/04-github/v1/05-returning-a-license.md
rename to versioned_docs/version-1/04-github/05-returning-a-license.md
diff --git a/docs/04-github/v1/06-deployment/android.md b/versioned_docs/version-1/04-github/06-deployment/android.md
similarity index 100%
rename from docs/04-github/v1/06-deployment/android.md
rename to versioned_docs/version-1/04-github/06-deployment/android.md
diff --git a/docs/04-github/v1/06-deployment/ios.md b/versioned_docs/version-1/04-github/06-deployment/ios.md
similarity index 100%
rename from docs/04-github/v1/06-deployment/ios.md
rename to versioned_docs/version-1/04-github/06-deployment/ios.md
diff --git a/versioned_docs/version-1/05-gitlab/01-getting-started.md b/versioned_docs/version-1/05-gitlab/01-getting-started.md
new file mode 100644
index 0000000..8db0260
--- /dev/null
+++ b/versioned_docs/version-1/05-gitlab/01-getting-started.md
```

The commit already does a _rename_. Something we could do is actually to remove `gitlab` and `intro` pages from v1 docs, but then I think it would become even more misleading.

webbertakken replied:
git usually recognises move operations. But if you have both files at the same time it will not recognise it as a move, even if you delete the file after that (depending on local git configuration).

Usually this only ever happens with 1~20 files. What I usually end up doing in a situation like yours is to generate the structure to see what the filenames should be and then on a file by file basis
- Copy the new filename. 
- Delete that new file. 
- Rename the original file to the copied file name

Then commit, and then make changes. Git recognises it's the same file until you make changes that affect more than a percentage of the file.

webbertakken replied:
Ah ok, so it's just because the GitLab files for v1 never existed before. Check.

GabLeRoux replied:
Considering this as resolved 👍

webbertakken commented:
> for now I suggest we keep the warning until we get rid of AntD and move to an actual mdx docs page with a component that fits docs styling 👍

Sounds good to me.

davidmfinol reviewed (approved):
This is exciting! I'm looking forward to seeing this deployed!

GabLeRoux commented on core/hooks/use-clipboard.tsx:
```diff
@@ -0,0 +1,22 @@
+export type OnSuccess = () => any;
+export type OnError = (error: Error) => any;
+
+export function useClipboard() {
```
Was this intended?

Any reason why we bring back `useClipboard`? This is built-in in docusaurus if I'm not mistaken. Docs page code blocks have it by default and we can reuse the code blocks in react pages as well just like I did here:
[`f95fc5c` (#259)](https://github.com/game-ci/documentation/pull/259/commits/f95fc5c93d58f54b9f084ebea735a60022e09772#diff-42eec9aff99056af77cdb3610b3ad8f2ef6c82b0696123159db1ade3af59acc8R66)

![ScreenShot Firefox Developer Edition 2022-05-17 at 17 31 24@2x](https://user-images.githubusercontent.com/1264761/168913679-fb54d11f-e501-4daf-81ae-2fae204f98f8.png)

See official documentation here:
https://docusaurus.io/docs/markdown-features/code-blocks#usage-in-jsx

webbertakken replied:
Yea it's used

![image](https://user-images.githubusercontent.com/20756439/168927906-593ebfce-257a-4ddc-87e8-a6c1f70ec1a3.png)

https://user-images.githubusercontent.com/20756439/168928173-4459e8cf-8cda-4dd9-8f3b-b2a85fee7ea2.mp4

GabLeRoux commented:
Unless we need `useClipboard`, I would remove it and consider this ready to merge 🎉
Maybe we can disable the algolia search until we finish its implementation with crawlers. We can create a new pr only for that as well. 

Right now, on the previous site, the current search stops working due to limits being reached from time to time so it wouldn't be that bad anyway. What do you think?

webbertakken commented:
> Unless we need `useClipboard`, I would remove it and consider this ready to merge 🎉 Maybe we can disable the algolia search until we finish its implementation with crawlers. We can create a new pr only for that as well.
> 
> Right now, on the previous site, the current search stops working due to limits being reached from time to time so it wouldn't be that bad anyway. What do you think?

Yea sounds good. Have you already requested Algolia DocSearch? Was it already approved?

webbertakken commented:
Updated your PR

- Linting works 
- and is enforced using lefthook
- Swizzled the Root element
- Added all global context providers (global state, toaster, database, auth, icons)
- Spinners, clipboard, notifications fixed

@GabLeRoux this probably needs another look to see if everything now works as expected but generally should be good to go after that. Feel free to move it forward 👍

GabLeRoux commented:
> > Unless we need `useClipboard`, I would remove it and consider this ready to merge 🎉 Maybe we can disable the algolia search until we finish its implementation with crawlers. We can create a new pr only for that as well.
> > Right now, on the previous site, the current search stops working due to limits being reached from time to time so it wouldn't be that bad anyway. What do you think?
> 
> Yea sounds good. Have you already requested Algolia DocSearch? Was it already approved?

Yes, we have everything approved for DocSearch. We can login here to manage the crawls:  
https://crawler.algolia.com/admin/users/login

The following configuration is currently setup to work with our approved docsearch algolia project:
[`docusaurus.config.js`'s `themeConfig` `algolia`](https://github.com/game-ci/documentation/pull/259/files#diff-a038096cbdea434999e1dce5ab497212f1fe18204dde1a027ce3bdd663261a2aR75-R97)

I think it's easier for now to isolate the docsearch integration in a future step. Here are the options I see right now:

1. Publish this pr as is, manually set the crawler config to the [docusaurus template](https://docsearch.algolia.com/docs/templates)  and run a manual crawl from the crawler webpage:
pros: easy, should work instantly and could be a good in-between step, we just need to try it
cons: won't work on all branches, will work only on the published website

2. Implement api calls reusing same logic as before, but invoking https://www.algolia.com/doc/rest-api/crawler/?client=json#create-a-crawler at some point during deployment. This would be the ideal way of doing it, but it also takwes more time to implement. Quoting you from a private discussion we had about the setup:
> - get crawlers
> - check if prefix from branch name exists (see slugify or something in our code)
>    - if exists, select that crawler
>    - if not exists: create crawler (id gets returned)
> - use crawler to index
> - poll until success on all pages from "count"

3. Disable docsearch and implement this in a different pr

GabLeRoux commented:
>     * Linting works
> 
>     * and is enforced using lefthook

I just tested that locally, I modified `src/pages/docs/docker/versions.tsx` with bad spaces and it got fixed so yes it works for me too here 👍 

<details>
<summary>Details on what I tried</summary>

```bash
yarn
echo "    " >> src/pages/docs/docker/versions.tsx
git add src/pages/docs/docker/versions.tsx
git commit -m "test"
```
The hook executed and fixed the file:
<img width="779" alt="ScreenShot iTerm2 2022-05-18 at 00 12 48@2x" src="https://user-images.githubusercontent.com/1264761/168956012-1c73a93f-3992-4428-b51b-11c3b5faf983.png">

</details>

On a side note, it took more time than what I expected. I think it can be improved later tho, better have this than not even if it takes some time.

<details>
<summary>Observations concerning eslint execution time</summary>
I tried it again and still getting similar result:

```
eslint '*/**/*.{js,ts,tsx}' --quiet --fix src/pages/docs/docker/versions.tsx
✨  Done in 11.36s.
```

Maybe it's just `eslint` that takes time and I'm not used to it. I tried these:

```bash
# similar to how it's invoked by lefthook
 time ./node_modules/.bin/eslint  '*/**/*.{js,ts,tsx}' --quiet --fix src/pages/docs/docker/versions.tsx
./node_modules/.bin/eslint '*/**/*.{js,ts,tsx}' --quiet --fix   17.17s user 1.48s system 157% cpu 11.832 total
```
```bash
# tried without '*/**/*.{js,ts,tsx}', seems to be a bit better
time ./node_modules/.bin/eslint --quiet --fix src/pages/docs/docker/versions.tsx
./node_modules/.bin/eslint --quiet --fix src/pages/docs/docker/versions.tsx  12.93s user 1.34s system 148% cpu 9.618 total
```

</details>

GabLeRoux commented:
I did a quick pass on each of the pages and everything seems fine. 

The only thing I found that might be worth fixing is that the `dracula` theme (used in dark theme) wasn't as nice as the `github` theme (used in light theme) or our previous one as we can see here:

Left side: docusaurus, right side, previous implementation.

<img width="1773" alt="ScreenShot Firefox Developer Edition 2022-05-18 at 01 05 59@2x" src="https://user-images.githubusercontent.com/1264761/168964800-2426286e-fab1-4f83-bda6-2adb8f19cf39.png">

<img width="1776" alt="ScreenShot Firefox Developer Edition 2022-05-18 at 01 05 25@2x" src="https://user-images.githubusercontent.com/1264761/168964805-8212be1c-d2c8-496f-bced-7b3a94aabb16.png">

My main complain about it is that most of the yaml keys aren't colored. I might be picky tho, but I use dracula theme every day in my IDE and my yaml files looks much better than the one from `FormidableLabs/prism-react-renderer` default dracula theme.

I went ahead with 57294e67b2cbc7240a4bf73f14476e5f376d7b40

Before:

<img width="1780" alt="ScreenShot Firefox Developer Edition 2022-05-18 at 02 25 31@2x" src="https://user-images.githubusercontent.com/1264761/168971739-eb8e1f12-4578-4fef-8559-e6180ea89f65.png">

After:

<img width="1779" alt="ScreenShot Firefox Developer Edition 2022-05-18 at 02 25 56@2x" src="https://user-images.githubusercontent.com/1264761/168971771-122565f8-5c58-47cc-b87c-3690e5157378.png">

The drak theme is also much closer to the `github` light theme.

GabLeRoux commented:
lgtm 🚀

GabLeRoux commented on docusaurus.config.js:
```diff
@@ -0,0 +1,334 @@
+
+const lightCodeTheme = require('prism-react-renderer/themes/github');
+
+const darkCodeTheme = {
```
I tried to put this in its own file, but I was getting errors like this:

```
[ERROR] /Users/gableroux/repos/gameci/game.ci/src/theme/dracula.js:110
export default theme;
^^^^^^

SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1033:15)
    at Module._compile (node:internal/modules/cjs/loader:1069:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/gableroux/repos/gameci/game.ci/docusaurus.config.js:8:23)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at module.exports (/Users/gableroux/repos/gameci/game.ci/node_modules/import-fresh/index.js:32:59)
    at loadSiteConfig (/Users/gableroux/repos/gameci/game.ci/node_modules/@docusaurus/core/lib/server/config.js:21:55)
error Command failed with exit code 1.
```

Couldn't figure it out, but putting the theme in here directly as an object works just fine. Looks a bit noisy, but this is also something we can improve later.

webbertakken replied:
note that in Docusaurus v2 the `theme` folder is a special folder intended for [Swizzling](https://docusaurus.io/docs/swizzling). Did you swizzle the theme or just put this file there?

GabLeRoux replied:
we fixed this in a later change by using `modules.export` instead of `export default` 👍

webbertakken commented:
> 

@GabLeRoux thanks for pointing that out. Fixing in https://github.com/game-ci/documentation/pull/272
